experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+  if isdefined( G, :name )
+    Base.print( io, G.name )
+    return
+  end
+  Base.println( io, "Glueing of" )
+  Base.println( io, G.firstPatch )
+  Base.println( io, "and" )
+  Base.println( io, G.secondPatch )
+  Base.println( io, "along" )
+  Base.println( io, domain(G.inclusionFirstPatch) )
+  return
+end
+
+function Base.copy( G::Glueing )
+  return Glueing( G.inclusionFirstPatch, G.inclusionSecondPatch, G.glueingIsomorphism )
+end
+
+################################################################################
+# Covering of a scheme.
+# It consists of a list of affine patches together with their glueings.
+# The glueings are stored in an array (which is usually half empty, since 
+# glueings need only be provided for ordered pairs of indices). 
+# The index methods for double arrays have been overloaded so that 
+# OrderedMultiindex can be used to address them.
+#
+@Markdown.doc """
+    mutable struct Covering
+
+Structure for storing a covering of a CoveredScheme X by affine 
+patches Uᵢ, i = 1,…,N. The necessary data consists of these affine 
+patches and the information on how to glue them pairwise via 
+glueins Gᵢⱼ: Uᵢ↩ Uᵢ∩ Uⱼ ↪ Uⱼ. 
+
+Once created, a covering must not be altered, since in general several 
+other objects, like e.g. vector bundles and morphisms of schemes, will 
+refer to such coverings and they need to rely on them to not change. 
+If one needs another covering derived from the given one, then a 
+new covering needs to be set up and stored. 
+"""
+mutable struct Covering
+  patches::Vector{AffineScheme}	# A list of open patches for this scheme
+  glueings::Matrix{Union{Glueing,Nothing}}     # A list of glueings between the patches 
+
+  # Variables for caching
+  name::String
+  						# listed above
+  function Covering( patches::Vector{AffineScheme}, glueings::Matrix{Union{Glueing,Nothing}} )
+    return new( patches, glueings )
+  end
+
+  function Covering()
+    C = new()
+    C.patches = AffineScheme[]
+    #C.glueings = Matrix{Union{Nothing,Glueing}}
+    return C
+  end
+
+  function Covering( X::AffineScheme ) 
+    C = new()
+    C.patches = [X]
+    C.glueings = Array{Nothing,2}(nothing,1,1)  # no glueing on the diagonal.
+    return C
+  end
+end
+
+function Base.copy( C::Covering )
+  # I'm afraid we need a copy here. Otherwise an alteration 
+  # of the original covering will also affect the new one.
+  return Covering( copy(C.patches), copy(C.glueings) )
+end
+
+@Markdown.doc """
+    function patches( C::Covering ) -> Vector{AffineScheme}
+
+Returns the patches of the covering.
+"""
+function patches( C::Covering )
+  if isdefined( C, :patches )
+    return C.patches
+  else 
+    return nothing 
+  end 
+end
+
+@Markdown.doc """
+    function patch( C::Covering, i::Int )
+    
+Returns the i-th patch in the list of patches for the covering C.
+"""
+function patch( C::Covering, i::Int )
+  if isdefined( C, :patches )
+    return( C.patches[i] )
+  else
+    return nothing
```

Comment:
Why not an out of bounds error? With the current code, `patch(C, -2)` or `patch(C, 2^60)` will work whenever there are no patches, but will fail once there are actually patches. That seems weird?

For that matter, why have this at all, when the user can also write `patches(C)[i]`?